src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkFeatureBand.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Deployment.DotNet.Releases;
+
+#nullable disable
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    public struct SdkFeatureBand : IEquatable<SdkFeatureBand>, IComparable<SdkFeatureBand>
+    {
+        private ReleaseVersion _featureBand;
+
+        public SdkFeatureBand(string version) : this(new ReleaseVersion(version) ?? throw new ArgumentNullException(nameof(version))) { }
+
+        public SdkFeatureBand(ReleaseVersion version)
+        {
+            var fullVersion = version ?? throw new ArgumentNullException(nameof(version));
+            if (string.IsNullOrEmpty(version.Prerelease) || version.Prerelease.Contains("dev") || version.Prerelease.Contains("ci"))
+            {
+                _featureBand = new ReleaseVersion(fullVersion.Major, fullVersion.Minor, fullVersion.SdkFeatureBand);
+            }
+            else
+            {
+                // Treat preview versions as their own feature bands
+                var prereleaseComponents = fullVersion.Prerelease.Split('.');
+                var formattedPrerelease = prereleaseComponents.Length > 1 ? 
```

Comment:
This assumes we'll never have more than 2 prerelease labels, e.g. -rc.2 or -preview.9

Reply:
@joeloff I'm not following.  The elements after the first two are the build numbers which we want to ignore for the purposes of feature bands.  We aren't expecting to have a Preview 3 and then a Preview 3.1, for example.

Reply:
Yup, that's fine, I just wanted to confirm the assumptions we're making because technically semver supports an infinite number of prerelease labels.